src/Tools/_framework/ToolPanels/GradebookStudent.jsx

```diff
@@ -141,18 +137,35 @@ export default function GradebookStudent() {
       }
 
       let numberScores = scores.length;
+
+      //Sort by points value and retain the maximumNumber
       scores = scores.sort((a, b) => b - a).slice(0, maximumNumber);
-      let categoryScore = scores.reduce((a, c) => a + c, 0) * scaleFactor;
-      let categoryAssignedPoints = allassignedpoints.reduce((a, c) => a + c, 0) * scaleFactor;
-      if (categoryAssignedPointsAreAllDashes){categoryAssignedPoints = '-'}
 
+      //Scale by scareFactor
+      let scaledScore = scores.reduce((a, c) => a + c, 0) * scaleFactor;
+      let scaledAssignedPoints =
```

Comment:
Before summing up allassignedpoints, you first need to sort by points value and retain the maximumNumber as done in 158 for allpossiblepoints.
(This flaw was introduced when assigned points were introduced, but we should correct it now.)

Reply:
That is there up above on 142

Reply:
Line 142 is for scores.  Need to also do it for scaledAssignPoints

Reply:
I.e. before 146, you want to add
```
      //Sort by points value and retain the maximumNumber
      allassignedpoints = allassignedpoints
        .sort((a, b) => b - a)
        .slice(0, maximumNumber);
```

Reply:
Otherwise, the assigned points will be too large, as it ignores maximumNumber.